src/components/VideoCard/index.jsx

```diff
@@ -77,7 +127,13 @@ const VideoCard = (props) => {
   } else {
     return (
       <div class="horizontal-card margin-trb-20">
-        <div class="image-container badge-container cursor-pointer" onClick={() => navigate(`/video/${youtubeID}`)}>
+        <div
+          class="image-container badge-container cursor-pointer"
+          onClick={() => {
+            addVideoToHistory(user.encodedToken, video);
+            navigate(`/video/${youtubeID}`);
+          }}
```

Comment:
suggestion: make a handler function to extract code logic from UI

Reply:
Sure . will look into it.